### PR TITLE
BUGFIX - fix some bugs

### PR DESCRIPTION
**Pull request by MaximeZmt**
## Fixed Bugs

* Unconnected user crash when finishing typing game
* Not updating the user status when user click on android built-in return arrow to come back to welcome screen

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/activity/WelcomeActivity.kt [hunk 1]
```diff
@@ -34,17 +34,24 @@ class WelcomeActivity : AppCompatActivity() {
         supportActionBar?.hide()
         setContentView(R.layout.activity_welcome_screen)
 
+        updateUserConnectionStatus()
+
+        Log.e("INFO: Should Access", ((InternetState.hasAccessedInternetOnce(this))).toString())
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/activity/WelcomeActivity.kt:39]:
```suggestion
```

File: app/src/main/java/ch/sdp/vibester/activity/WelcomeActivity.kt [hunk 2]
```diff
@@ -34,17 +34,24 @@ class WelcomeActivity : AppCompatActivity() {
...
+        PersistanceSetter.setPersistance()
+        Database.get()
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/activity/WelcomeActivity.kt:41]: why do you need Database.get() here if the result is not saved anywhere?

**Review by kamilababayeva — APPROVED** [review]
Thanks!! Nice fixes! I have left two comments, but otherwise all good!

**Review by jiabaow — APPROVED** [review]
LGTM

**Pull request merged**